src/uct/ib/mlx5/dv/ib_mlx5_dv.c

```diff
@@ -278,6 +293,11 @@ void uct_ib_mlx5_devx_destroy_qp(uct_ib_mlx5_qp_t *qp)
     if (qp->devx.mem != NULL) {
         mlx5dv_devx_umem_dereg(qp->devx.mem);
     }
+
+    if ((qp->devx.wq_buf != NULL) && uct_ib_fork_enabled) {
+        madvise(qp->devx.wq_buf, qp->devx.wq_len, MADV_DOFORK);
+    }
```

Comment:
maybe wrap madvise in helper function which will also accept log_level to use when reporting error

Reply:
maybe add ucs_posix_memalign/madvise/mlx5dv_devx_umem_reg helper (and dereg counterpart), add struct to aggregate  mlx5dv_devx_umem and size and also use size!=0 as flag for fork_enabled in destroy flow?

Reply:
yep, uct_ib_mlx5_buf_alloc